qiskit/circuit/library/standard_gates/rz.py

```diff
@@ -56,23 +58,20 @@ class RZGate(Gate):
         `1612.00858 <https://arxiv.org/abs/1612.00858>`_
     """
 
-    def __init__(self, phi, label=None):
+    def __init__(self, phi, phase=0, label=None):
         """Create new RZ gate."""
-        super().__init__('rz', 1, [phi], label=label)
+        super().__init__('rz', 1, [phi], phase=phase, label=label)
 
     def _define(self):
         """
         gate rz(phi) a { u1(phi) a; }
         """
         from .u1 import U1Gate
-        definition = []
         q = QuantumRegister(1, 'q')
-        rule = [
-            (U1Gate(self.params[0]), [q[0]], [])
+        phase = self.phase - 0.5 * self.params[0]
+        self.definition = [
+            (U1Gate(self.params[0], phase=phase), [q[0]], [])
         ]
```

Comment:
here is where i would make it from a U1 but set it to be a phase that to make it Rz